src/main/cms/cms.c

```diff
@@ -358,13 +358,14 @@ static void cmsPadToSize(char *buf, int size)
 
 static int cmsDisplayWrite(displayPort_t *instance, uint8_t x, uint8_t y, uint8_t attr, const char *s)
 {
-    uint8_t *c = (uint8_t*)s;
-    const uint8_t *cEnd = c + strlen(s);
-    for (; c != cEnd; c++) {
-        *c = toupper(*c);  // uppercase only
-        *c = (*c < 0x20 || *c > 0x5F) ? ' ' : *c; // limit to alphanumeric and punctuation 
+    char buffer[40];
+    unsigned index = 0;
+    for (; index < strlen(s); index++) {
```

Comment:
```suggestion
    uint8_t index = 0;
    uint8_t len = strlen(s);
    for (; index < len; index++) {
```
But maybe the compiler optimizes this anyway.

Reply:
gcc will generally produce worse code with `uint8_t` instead of `unsigned` (sign/zero extend instructions are emitted even when not needed).
`strlen()` should be optimized (moved outside the loop), but it seems that gcc 8 does not do it (clang does, but unrolls the code a lot).

The code can avoid strlen completely:
```
static int cmsDisplayWrite(displayPort_t *instance, uint8_t x, uint8_t y, uint8_t attr, const char *s)
{
   char buffer[40];
   char* b = buffer;
   while (*s) {
       char c = toupper(*s++);
       *b++ = (c < 0x20 || c > 0x5F) ? ' ' : c; // limit to alphanumeric and punctuation
    }
    *b++='\0';
    return displayWrite(instance, x, y, attr, buffer);
}

Reply:
BTW: https://godbolt.org/z/bq4v9T

Reply:
@ledvinap: This was done after 1.5 hours of bisecting at 2 am, it wasn't my best. Thanks for the cleanup.